uefi-test-runner/src/proto/console/stdout.rs

```diff
@@ -10,48 +10,71 @@ pub fn test(stdout: &mut Output) {
 
     // Print all modes.
     for (index, mode) in stdout.modes().enumerate() {
+        let mode = mode.expect("Warnings encountered while querying text mode");
         info!(
-            "- Graphics mode #{}: {} rows by {} columns",
+            "- Text mode #{}: {} rows by {} columns",
             index,
             mode.rows(),
             mode.columns()
         );
     }
 
     // Should clean up after us.
-    stdout.reset(false).unwrap();
+    stdout.reset(false).unwrap().unwrap();
 }
 
-// Switch to the maximum supported graphics mode.
+// Switch to the maximum supported text mode.
 fn change_text_mode(stdout: &mut Output) {
-    let best_mode = stdout.modes().last().unwrap();
+    let best_mode = stdout
+        .modes()
+        .last()
+        .unwrap()
+        .expect("Warnings encountered while querying text mode");;
     stdout
         .set_mode(best_mode)
-        .expect("Failed to change graphics mode");
+        .expect("Failed to change text mode")
+        .expect("Warnings encountered while changing text mode");
 }
 
 // Set a new color, and paint the background with it.
 fn change_color(stdout: &mut Output) {
     stdout
         .set_color(Color::White, Color::Blue)
-        .expect("Failed to change console color");
-    stdout.clear().expect("Failed to clear screen");
+        .expect("Failed to change console color")
+        .expect("Warnings encountered while changing console color");
+    stdout
+        .clear()
+        .expect("Failed to clear screen")
+        .expect("Warnings encountered while clearing screen");
 }
 
 // Print a text centered on screen.
 fn center_text(stdout: &mut Output) {
     // Move the cursor.
     // This will make this `info!` line below be (somewhat) centered.
-    stdout.enable_cursor(true).unwrap_or_else(|s| match s {
-        Status::UNSUPPORTED => info!("Cursor visibility control unavailable"),
-        _ => panic!("Failed to show cursor"),
-    });
+    stdout
+        .enable_cursor(true)
+        .unwrap_or_else(|s| {
+            match s {
+                Status::UNSUPPORTED => info!("Cursor visibility control unavailable"),
```

Comment:
hmm, we should eventually refactor these tests to return a `Result` instead of manually checking for status codes everywhere.

Reply:
Yeah, I was thinking about using `Result` _just_ for these leaf / terminal test functions. At least until we get a pure Rust backtrace library.